distributed/tests/test_client.py

```diff
@@ -8029,3 +8030,41 @@ def identity(x):
     outer_future = c.submit(identity, {"x": inner_future, "y": 2})
     result = await outer_future
     assert result == {"x": 1, "y": 2}
+
+
+def bad_wrapper(client, use_worker_client=False):
+    class FuncWrapper:
+        def __init__(self, f, args=None):
+            self.f = f
+            self.args = [] if args is None else args
+
+        def __call__(self, x):
+            return self.f(x, *self.args)
+
+    def make_data():
+        return 1
+
+    def func_to_map(x, x0):
+        if use_worker_client:
+            ctx = worker_client()
+        else:
+            ctx = nullcontext()
+        with ctx:
+            return x0.result() + x
+
+    shared_future = client.submit(make_data)
+
+    wrapped_func = FuncWrapper(func_to_map, (shared_future,))
+
+    futures = client.map(wrapped_func, range(10))
+    results = client.gather(futures)
+
+
+@pytest.mark.slow()
+def test_cancelled_error_wrapped_future(loop):
+    # See https://github.com/dask/distributed/issues/7746
+    with LocalCluster(loop=loop, dashboard_address=":0", n_workers=1) as cluster:
```

Comment:
```suggestion
    with LocalCluster(loop=loop, dashboard_address=":0", n_workers=1, threads_per_worker=2) as cluster:
```
The number of threads used here would vary based on the machine, right? I could only get https://github.com/dask/distributed/issues/7746 to reproduce reliably with >=2 threads _and_ a small sleep in `func_to_map`.